### PR TITLE
New authentication methods; remove pester in favor of faraday based retry

### PR DESCRIPTION
**Pull request by apurvis**
should fix https://github.com/jarthod/survey-gizmo-ruby/issues/62 and https://github.com/jarthod/survey-gizmo-ruby/issues/60 @jarthod

### CODE REVIEW

**Comment by apurvis** [comment]
@jarthod  i'm gonna wait at least a few days and watch for stability issues before merging this

File: lib/survey_gizmo/logger.rb [hunk 1]
```diff
@@ -3,6 +3,9 @@
 module SurveyGizmo
   class Logger < ::Logger
     def format_message(severity, timestamp, progname, msg)
+      msg.gsub!(/#{Regexp.quote(SurveyGizmo.configuration.api_token_secret)}/, '<SG_API_KEY>') if SurveyGizmo.configuration.api_token
```
Comment (jarthod) [inline lib/survey_gizmo/logger.rb]: bad copy/paste I guess, should be `api_token` here, not `api_token_secret`

**Comment by jarthod** [comment]
Looks great, and removes a dependency in favor of native faraday stuff, cool :+1:

File: lib/survey_gizmo/logger.rb (hunk 1 above)
Comment (apurvis) [inline lib/survey_gizmo/logger.rb]: nice catch thx

**Comment by apurvis** [comment]
so far looks stable; whenever you want to merge & release (i would merge but prolly just better for you to merge when you're ready to do a release, given that this is a release everyone is going to need to upgrade to)

**Comment by jarthod** [comment]
Great, thanks.
I'll merge release this next week.

**Comment by jarthod** [comment]
Released `6.0.2`

**Pull request merged**